### PR TITLE
Upstream/release purview azure.analytics.purview.catalog 2022 08 01 preview

### LINKED ISSUE #22185: [Microsoft Purview - Catalog] API Review
New API Review meeting has been requested.

**Service Name**: Microsoft Purview - Catalog
**Review Created By**: Yifan Zhou
**Review Date**: 2/2/2023 4:00 PM PT
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/21770
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: https://review.learn.microsoft.com/en-us/rest/api/documentation-preview/glossary/get-bulk-delete-operation-status?view=azure-rest-preview&branch=result_openapiHub_production_4ed3c9e48b9f&tabs=HTTP

**Description**: This PR includes adding two new APIs and some other API updates like updating descriptions/examples/response properties etc.

Updated paths:
Delete /glossary/terms
Get /glossary/terms/bulkDeletion/{operationGuid}

Documentation preview: 
https://review.learn.microsoft.com/en-us/rest/api/documentation-preview/glossary/get-bulk-delete-operation-status?view=azure-rest-preview&branch=result_openapiHub_production_4ed3c9e48b9f&tabs=HTTP

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=5c5df432-f992-ed11-aad1-000d3a3418aa)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by yifan-zhou922**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-03-01-preview/purviewcatalog.json
* Updated paths:
Delete /glossary/terms
Get /glossary/terms/bulkDeletion/{operationGuid}

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

fix https://github.com/Azure/azure-rest-api-specs/issues/22185

### CODE REVIEW

**Comment by raych1** [comment]
@yifan-zhou922 , the `ModelValidation` errors are all about the type mismatch. when the property is annotated as https://github.com/Azure/azure-rest-api-specs/blob/a9e13fe529181bd92099beaffbd9d20a9b6409ee/specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json#L6613 , it means the value of each element in the map should be object type. It cannot be a string or primitive types.

**Comment by yifan-zhou922** [comment]
Recommendations for updating async error schema and url path for bulk delete glossary terms have been deployed to Production.

image

image

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json [hunk 1]
```diff
@@ -0,0 +1,8364 @@
...
+            "$ref": "./examples/Glossary_GetBulkDeleteOperationStatus.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/forceDeleteEntityAssignment"
+          },
+          {
+            "$ref": "#/parameters/forceDeleteChild"
+          },
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json:3089]: nit, one thing to think about boolean is whether one day you'd wish to have an enum here for other than true/false.

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json [hunk 2]
```diff
@@ -0,0 +1,8364 @@
...
+      "description": "Enum of the status of import csv operation."
+    },
+    "BulkDeleteOperation": {
+      "type": "object",
+      "title": "Status of bulk delete operation",
+      "description": "The status of bulk delete operation",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "guid string"
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json:7768]: nit, `guid string` is a pretty useless description...

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json [hunk 3]
```diff
@@ -0,0 +1,8364 @@
...
+          "$ref": "#/definitions/BulkDeleteOperationStatus"
+        },
+        "createTime": {
+          "type": "string",
+          "description": "The created time of the record."
+        },
+        "lastUpdateTime": {
+          "type": "string",
+          "description": "The last updated time of the record."
+        },
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json:7780]: format data-time?

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json [hunk 4]
```diff
@@ -0,0 +1,8364 @@
...
+        "properties": {
+          "x-ms-client-flatten": true,
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json:7782]: It is not mgmt. Do we like client flatten here?

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json [hunk 5]
```diff
@@ -0,0 +1,8364 @@
...
+        },
+        "createTime": {
+          "type": "string",
+          "description": "The created time of the record."
+        },
+        "lastUpdateTime": {
+          "type": "string",
+          "description": "The last updated time of the record."
+        },
+        "properties": {
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json:7781]: I hear that is deployed.

But I remember recommendation is "result"?

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json [hunk 6]
```diff
@@ -0,0 +1,8364 @@
...
+        "modelAsString": true
+      },
+      "title": "ImportCSVOperationStatus",
+      "description": "Enum of the status of import csv operation."
+    },
+    "BulkDeleteOperation": {
+      "type": "object",
+      "title": "Status of bulk delete operation",
+      "description": "The status of bulk delete operation",
+      "properties": {
```
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json:7765]: Consider add `required`. I assume `id` and `status` will always returned?

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json (hunk 5 above)
Comment (yifan-zhou922) [reply specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json:7781]: We only changed error schema in this round according to the recommendation. The rest remains unchanged.

File: specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json (hunk 2 above)
Comment (yifan-zhou922) [reply specification/purview/data-plane/Azure.Analytics.Purview.Catalog/preview/2022-08-01-preview/purviewcatalog.json]: useless description removed.

**Comment by yifan-zhou922** [comment]
Hi @mikekistler / @markweitzel , could you please help review and signoff the changes? We have updated the PR according to the comments in the review meeting. Many thanks.

**Comment by weidongxu-microsoft** [comment]
First and foremost, please make your CI pass.

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

Since this is based on an industry API, we can't offer much actionable guidance.

**Pull request merged**